llvm/lib/Target/AMDGPU/GCNSubtarget.h

```diff
@@ -1315,6 +1315,10 @@ class GCNSubtarget final : public AMDGPUGenSubtargetInfo,
   // of sign-extending.
   bool hasGetPCZeroExtension() const { return GFX12Insts; }
 
+  // \returns true if the target supports signed immediate offset for SMRD
+  // instructions.
+  bool hasSignedSMRDImmOffset() const { return getGeneration() >= GFX9; }
```

Comment:
We already have a static `hasSMRDSignedImmOffset` in AMDGPUBaseInfo. Can you just expose that instead?